### ISSUE TITLE
Inspec Installation on RHEL and CentOS 7 Fails with a Signing Error

### ISSUE THREAD

**Issue by tohch4** [darkwizard242/ansible-role-inspec#6]
##### SUMMARY
I work with RedHat Enterprise Linux 7 gold images that require [DISA STIG hardening requirements for `yum` and other components not part of the defaults](https://www.stigviewer.com/stig/red_hat_enterprise_linux_7/2017-12-14/finding/V-71981). For this reason, when I run the current release version of `master` for this module via Ansible Galaxy, installation fails because this package does not preemptively add, for the tasks and defaults targeting RedHat/EL as the `ansible_os_dist`, the [Chef Signing key for RPM package](https://docs.chef.io/packages/). Therefore, the package downloads to a temporary directory, but fails to execute the package installation.

##### ISSUE TYPE

Without modification, this role fails with current defaults with the an error.

If package signing is enabled before this role with a `pre_tasks` task, it works without error.

##### ANSIBLE VERSION

```sh
$ ansible --version
ansible 2.9.11
  config file = None
  configured module search path = ['/Users/myusername/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.9.11/libexec/lib/python3.8/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.5 (default, Jul 21 2020, 10:42:08) [Clang 11.0.0 (clang-1100.0.33.17)]
```

##### CONFIGURATION

```
$ ansible-config dump --only-changed
$ echo $?
0
```

##### OS / ENVIRONMENT

```
Darwin mycomputername 19.6.0 Darwin Kernel Version 19.6.0: Thu Jun 18 20:49:00 PDT 2020; root:xnu-6153.141.1~1/RELEASE_X86_64 x86_64
```
##### STEPS TO REPRODUCE

With error:

```yaml
# playbook.yml
---
    - hosts: all
      become: yes

      roles:
        - epel
        - inspec
```

```yaml
# requirements.yml

- src: git+https://githubent.private.tld/org/ansible-install_docker_ce.git
  version: master
  name: docker_ce

- src: darkwizard242.inspec
  version: master
  name: inspec
```

Without error:

```yaml
# playbook.yml
---
    - hosts: all
      become: yes

      pre_tasks:
        - name: Download Chef signing key for RPM packages
          rpm_key:
            key: https://packages.chef.io/chef.asc
            fingerprint: 1168 5DB9 2F03 640A 2FFE 7CA8 2940 ABA9 83EF 826A
            state: present

      roles:
        - epel
        - inspec
```

```yaml
# requirements.yml

- src: git+https://githubent.private.tld/org/ansible-install_docker_ce.git
  version: master
  name: docker_ce

- src: darkwizard242.inspec
  version: master
  name: inspec
```

##### EXPECTED RESULTS

It installs without GPG signature checks with errors.

##### ACTUAL RESULTS

```sh
amazon-ebs: TASK [EL Family | Downloading and extracting inspec 4.22.22] *******************
amazon-ebs: fatal: [default]: FAILED! => {"changed": false, "changes": {"installed": ["/home/ec2-user/~myusername/.ansible/tmp/ansible-tmp-1599687345.885809-52020-25354355402861/inspec-4.22.22-1.el7.x86_64cM9glv.rpm"]},
"msg": "warning: /home/ec2-user/~myusername/.ansible/tmp/ansible-tmp-1599687345.885809-52020-25354355402861/inspec-4.22.22-1.el7.x86_64cM9glv.rpm:
Header V4 DSA/SHA1 Signature, key ID 83ef826a: NOKEY\n\n\nPublic key for inspec-4.22.22-1.el7.x86_64cM9glv.rpm is not installed\n", "rc": 1,
"results": ["Loaded plugins: amazon-id, s3iam, search-disabled-repos\nExamining /home/ec2-user/~myusername/.ansible/tmp/ansible-tmp-1599687345.885809-52020-25354355402861/inspec-4.22.22-1.el7.x86_64cM9glv.rpm: 
inspec-4.22.22-1.el7.x86_64\nMarking /home/ec2-user/~myusername/.ansible/tmp/ansible-tmp-1599687345.885809-52020-25354355402861/inspec-4.22.22-1.el7.x86_64cM9glv.rpm to be installed\nResolving
Dependencies\n--> Running transaction check\n---> Package inspec.x86_64 0:4.22.22-1.el7 will be installed\n--> Finished Dependency Resolution\n--> Finding unneeded leftover dependencies\n
Found and removing 0 unneeded dependencies\n\nDependencies Resolved\n\n================================================================================\n
Package  Arch     Version           Repository                            Size\n
================================================================================\n
Installing:\n inspec   x86_64   4.22.22-1.el7     /inspec-4.22.22-1.el7.x86_64cM9glv   168 M\n\nTransaction Summary\n
================================================================================\nInstall  1 Package\n\nTotal size: 168 M\nInstalled size: 168 M\nDownloading packages:\n"]}
```

##### SCREENSHOTS

<img width="1104" alt="image" src="https://user-images.githubusercontent.com/39844334/92673263-b74f8d80-f2e8-11ea-8b8a-6d7751669f58.png">

**Comment by tohch4** [comment]
I am totally call with committing additional code for this in a PR, can I add it?

**Comment by darkwizard242** [comment]
@tohch4 - Understandable, I had not developed it with STIG'd systems in mind.. so didn't test them out on any hardened OS. 

Sure, please go ahead and create a PR with changes. There should be an automated TravisCI build that would kick off of your PR to test out the changes using molecule.

**Comment by tohch4** [comment]
Sorry for the delay, I have not forgotten, I will try to take care of this later this weekend.

**Comment by darkwizard242** [comment]
Sure @tohch4 , sounds good 👍🏻 and thanks very much for both finding the issue, reporting it and developing it's solution.

**Comment by tohch4** [comment]
I guess my fixes notation had a word between that and issue number, so I can close now #7 is merged. Thanks again!

**Comment by darkwizard242** [comment]
@tohch4 - All good, released version [1.1.3](https://github.com/darkwizard242/ansible-role-inspec/releases/tag/1.1.3). TravisCI will automatically push new tag version to Ansible Galaxy for darkwizard242.inspec role once the builds complete successfully.

Once again, thanks a lot mate 👍🏻
